<launch>

  <node name="naval_frame" pkg="baxterkitchen" type="naval_frame.py" respawn="false" output="screen" />

  
  <arg name="marker_size" default="5.6" />
  <arg name="max_new_marker_error" default="0.08" />
  <arg name="max_track_error" default="0.2" />

  <arg name="cam_image_topic" default="/usb_cam/image_raw" />

  <arg name="cam_info_topic" default="/usb_cam/camera_info" />

  <arg name="output_frame" default="/usb_cam" />


  <node name="usb_cam" pkg="usb_cam" type="usb_cam_node"  output="screen" >
    <param name="video_device" value="/dev/video0" />

    <!-- Parameters for Microsoft cameras -->

    <param name="image_width" value="1280" />
    <param name="image_height" value="720" />
    <param name="camera_info_url" value="file:///home/cc/ee106a/fa16/class/ee106a-acu/ee206akitchen/src/baxterkitchen/yml/lifecam.yml" />
    <!-- <param name="camera_info_url" value="file:///home/team15/ee206akitchen/src/baxterkitchen/yml/lifecam.yml" /> -->


    <!-- Parameters for Logitech cameras -->
<!--
    <param name="image_width" value="640" />
    <param name="image_height" value="472" />
    <param name="camera_info_url" value="file:///home/cc/ee106a/fa16/class/ee106a-acu/ee206akitchen/src/baxterkitchen/yml/usb_cam.yml" />
-->

    <param name="pixel_format" value="mjpeg" />
    <param name="camera_frame_id" value="usb_cam" />
    <param name="io_method" value="mmap" />
  </node>

  <arg name="bundle_files" default="$(find ar_track_alvar)/bundles/knife_0_1_2.xml $(find ar_track_alvar)/bundles/carrot_6_7_8.xml $(find ar_track_alvar)/bundles/naval_15_17_16.xml $(find ar_track_alvar)/bundles/corn_3_4_5.xml $(find ar_track_alvar)/bundles/sponge_9_10_11.xml $(find ar_track_alvar)/bundles/dish_12_13_19.xml $(find ar_track_alvar)/bundles/rack_14_18.xml" />
  
  <node name="ar_track_alvar" pkg="ar_track_alvar" type="findMarkerBundlesNoKinect" respawn="false" output="screen" args="$(arg marker_size) $(arg max_new_marker_error) $(arg max_track_error) $(arg cam_image_topic) $(arg cam_info_topic) $(arg output_frame) $(arg bundle_files)" />


</launch>
